src/charm.py

```diff
@@ -83,76 +97,81 @@ def __init__(self, *args):
+            self._on_config_changed,
         )
 
-        # TODO: Register Identity Platform Event Handlers
+        self.framework.observe(
+            self.on[KRATOS_ENDPOINTS_INTEGRATION_NAME].relation_changed,
+            self._on_config_changed,
+        )
 
-    # Event Handlers
-    # Charm event handlers"""
+        self.framework.observe(
+            self.loki_consumer.on.promtail_digest_error,
+            self._promtail_error,
+        )
 
     def _on_admin_ui_pebble_ready(self, event: WorkloadEvent) -> None:
         """Define and start a workload using the Pebble API."""
-        if not self._container.can_connect():
-            event.defer()
-            self.unit.status = WaitingStatus("Waiting to connect to admin-ui container")
-            return
+        self.unit.open_port(protocol="tcp", port=ADMIN_UI_PORT)
 
-        # Makes sure the directory for the logfile exists
-        if not self._container.isdir(str(LOG_DIR)):
-            self._container.make_dir(path=str(LOG_DIR), make_parents=True)
-            logger.info(f"Created directory {LOG_DIR}")
+        self._handle_status_update_config(event)
 
         self._set_version()
-        self._update_pebble_layer(event)
-
-    def _on_install(self, event: InstallEvent) -> None:
-        if not self._container.can_connect():
-            event.defer()
-            logger.info("Cannot connect to admin-ui container. Deferring the event.")
-            self.unit.status = WaitingStatus("Waiting to connect to admin-ui container")
-            return
-
-        # Makes sure the directory for the logfile exists
-        # Duplicated to avoid race condition between install and pebble_ready events.
-        if not self._container.isdir(LOG_DIR):
-            self._container.make_dir(path=LOG_DIR, make_parents=True)
-            logger.info(f"Created directory {LOG_DIR}")
 
     def _on_config_changed(self, event: ConfigChangedEvent) -> None:
         """Handle changed configuration."""
-        self._update_pebble_layer(event)
+        self._handle_status_update_config(event)
+
+    def _on_ingress_ready(self, event: IngressPerAppReadyEvent) -> None:
+        if self.unit.is_leader():
+            logger.info("This app's public ingress URL: %s", event.url)
+
+    def _on_ingress_revoked(self, event: IngressPerAppRevokedEvent) -> None:
+        if self.unit.is_leader():
+            logger.info("This app no longer has ingress")
 
-    def _update_pebble_layer(self, event: HookEvent) -> None:
+    def _handle_status_update_config(self, event: HookEvent) -> None:
         if not self._container.can_connect():
             event.defer()
             logger.info("Cannot connect to admin-ui container. Deferring the event.")
             self.unit.status = WaitingStatus("Waiting to connect to admin-ui container")
             return
 
-        self.unit.status = MaintenanceStatus("Configuration in progress")
+        if not self.model.relations[KRATOS_ENDPOINTS_INTEGRATION_NAME]:
```

Comment:
Perhaps the lib should provide a helper function for this

Reply:
thanks, I'll address that in the kratos PR